### PR TITLE
[SYCL] Move host accessor details inside the library

### PR DESCRIPTION
**Pull request by alexbatashev**
Signed-off-by: Alexander Batashev <alexander.batashev@intel.com>

### CODE REVIEW

File: sycl/include/CL/sycl/detail/accessor_impl.hpp [hunk 1]
```diff
@@ -64,20 +63,11 @@ class AccessorImplHost {
   AccessorImplHost(id<3> Offset, range<3> AccessRange, range<3> MemoryRange,
                    access::mode AccessMode, detail::SYCLMemObjI *SYCLMemObject,
                    int Dims, int ElemSize, int OffsetInBytes = 0,
-                   bool IsSubBuffer = false)
-      : MOffset(Offset), MAccessRange(AccessRange), MMemoryRange(MemoryRange),
-        MAccessMode(AccessMode), MSYCLMemObj(SYCLMemObject), MDims(Dims),
-        MElemSize(ElemSize), MOffsetInBytes(OffsetInBytes),
-        MIsSubBuffer(IsSubBuffer) {}
+                   bool IsSubBuffer = false);
```
Comment (romanovvlad) [inline sycl/include/CL/sycl/detail/accessor_impl.hpp]: Can we move entire AccessorImplHost  class to source dir?

**Comment by mihails-strasuns-intel** [comment]
Verified GDB accessor tests (with a host device) using the following patch:
```diff
diff --git a/sycl/xmethods/libsycl.so-gdb.py b/sycl/xmethods/libsycl.so-gdb.py
index a2c1fc810b7..930fe526e1c 100644
--- a/sycl/xmethods/libsycl.so-gdb.py
+++ b/sycl/xmethods/libsycl.so-gdb.py
@@ -18,9 +18,10 @@ class Accessor:
     def data(self):
         pass
 
-    def __init__(self, obj, result_type, depth):
+    def __init__(self, obj, result_type, class_type, depth):
         self.obj = obj
         self.result_type = result_type
+        self.class_type = class_type
         self.depth = depth
 
     def index(self, arg):
@@ -42,17 +43,19 @@ class Accessor:
 For Host device memory layout
 """
 class HostAccessor(Accessor):
-    def payload(self):
-        return self.obj['impl']['_M_ptr'].dereference()
+    def call_method(self, name):
+        return gdb.parse_and_eval(
+            "((%s*) %s)->%s()" % (self.class_type, self.obj, name));
 
     def memory_range(self, dim):
-        return self.payload()['MMemoryRange']['common_array'][dim]
+        return self.call_method('getMemoryRange')['common_array'][dim]
 
     def offset(self, dim):
-        return self.payload()['MOffset']['common_array'][dim]
+        return self.call_method('getOffset')['common_array'][dim]
 
     def data(self):
-        return self.payload()['MData']
+        return self.call_method('getPtr')
+
 
 """
 For CPU/GPU memory layout
@@ -87,8 +90,8 @@ class AccessorOpIndex(gdb.xmethod.XMethodWorker):
         # No way to wasily figure out which devices is currently being used,
         # try all accessor implementations until one of them works:
         accessors = [
-            DeviceAccessor(obj, self.result_type, self.depth),
-            HostAccessor(obj, self.result_type, self.depth)
+            DeviceAccessor(obj, self.result_type, self.class_type, self.depth),
+            HostAccessor(obj, self.result_type, self.class_type, self.depth)
         ]
         for accessor in accessors:
             try:
```

**Comment by bader** [comment]
@alexbatashev, please, resolve merge conflicts.

**Comment by bader** [comment]
@romanovvlad, ping.

**Comment by bader** [comment]
@romanovvlad, could you take a review this change, please?

**Pull request closed without merging**